modules/core/src/main/java/org/apache/ignite/IgniteSystemProperties.java

```diff
@@ -987,6 +987,15 @@
      */
     public static final String IGNITE_DISABLE_WAL_DURING_REBALANCING = "IGNITE_DISABLE_WAL_DURING_REBALANCING";
 
+    /**
+     * When property is set {@code true} each next exchange will try to compare with previous.
+     * If last rebalance is equivalent with new possible one, new rebalance does not trigger.
+     * Set the property {@code false} and each exchange will try to trigger new rebalance.
+     *
+     * Default is {@code false}.
+     */
+    public static final String IGNITE_REBALANCE_SCHEDULING_OPTIMIZATION = "IGNITE_REBALANCE_SCHEDULING_OPTIMIZATION";
```

Comment:
Better to name it
IGNITE_DISABLE_REBALANCING_CANCELLATION_OPTIMIZATION
and change the logic accordingly (true by default)

Reply:
I leaved comment above.